src/test/regress/sql/multi_row_router_insert.sql

```diff
@@ -0,0 +1,41 @@
+\set VERBOSITY terse
+
+SET citus.next_shard_id TO 1511000;
+SET citus.shard_replication_factor TO 1;
+SET citus.enable_local_execution TO ON;
+SET citus.log_local_commands TO ON;
+
+CREATE SCHEMA multi_row_router_insert;
+SET search_path TO multi_row_router_insert;
+
+SET client_min_messages to ERROR;
+SELECT 1 FROM master_add_node('localhost', :master_port, groupId => 0);
+RESET client_min_messages;
+
+-- when using local execution, multi-row & router inserts works fine
+-- even when not specifying some default columns
+
+CREATE TABLE reference_table(a INT DEFAULT 1111, b INT DEFAULT 2222);
+SELECT create_reference_table('reference_table');
+
+INSERT INTO reference_table VALUES (5), (6);
+INSERT INTO reference_table VALUES (DEFAULT), (7);
+INSERT INTO reference_table (b) VALUES (8), (9);
+
+SELECT * FROM reference_table ORDER BY 1,2;
+
+CREATE OR REPLACE FUNCTION square(a INT) RETURNS INT AS $$
+BEGIN
+    RETURN a*a;
+END; $$ LANGUAGE PLPGSQL STABLE;
+
+CREATE TABLE citus_local_table(a int, b int DEFAULT square(10));
```

Comment:
These examples touches local plan execution code, and so it'd be nice to test them with (a) prepared statements (b) actual stable  functions that would give different results per statement. You can check (and use) the builtin functions: `select proname from pg_proc where provolatile = 's';`